WallsAndHoles/tilemaphelpers.h

```diff
@@ -0,0 +1,37 @@
+#ifndef TILEMAPHELPERS_H
+#define TILEMAPHELPERS_H
+
+#include "tilemap.h"
+
+#include <QStack>
+#include <QSet>
```

Comment:
Since you're not using QSets and QStacks in this header, I would move these includes into the cpp file.